pkg/gofr/datasource/dgraph/draph.go

```diff
@@ -0,0 +1,252 @@
+package dgraph
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"google.golang.org/grpc/credentials/insecure"
+
+	"github.com/dgraph-io/dgo/v210"
+	"github.com/dgraph-io/dgo/v210/protos/api"
+	"google.golang.org/grpc"
+)
+
+// Config holds the configuration for connecting to Dgraph.
+type Config struct {
+	Host string
+	Port string
+}
+
+// Client represents the Dgraph client with logging and metrics.
+type Client struct {
+	client  DgraphClient
+	conn    *grpc.ClientConn
+	logger  Logger
+	metrics Metrics
+	config  Config
+}
+
+type (
+	Mutation  = api.Mutation
+	Operation = api.Operation
+)
+
+var (
+	errInvalidMutation   = errors.New("invalid mutation type")
+	errInvalidOperation  = errors.New("invalid operation type")
+	errHealthCheckFailed = errors.New("dgraph health check failed")
+)
+
+// New creates a new Dgraph client with the given configuration.
+func New(config Config) *Client {
+	return &Client{
+		config: config,
+	}
+}
+
+// Connect connects to the Dgraph database using the provided configuration.
+func (c *Client) Connect() {
+	address := fmt.Sprintf("%s:%s", c.config.Host, c.config.Port)
+	c.logger.Logf("connecting to dgraph at %v", address)
+
+	conn, err := grpc.Dial(address, grpc.WithTransportCredentials(insecure.NewCredentials()))
+	if err != nil {
+		c.logger.Errorf("error connecting to Dgraph, err: %v", err)
+		return
+	}
+
+	c.logger.Logf("connected to dgraph client at %v:%v", c.config.Host, c.config.Port)
+
+	// Register metrics
+	// Register all metrics
+	c.metrics.NewHistogram("dgraph_query_duration", "Response time of Dgraph queries in milliseconds.",
+		0.05, 0.1, 0.2, 0.5, 1, 2, 5, 10)
+	c.metrics.NewHistogram("dgraph_query_with_vars_duration", "Response time of Dgraph queries with variables in milliseconds.",
+		0.05, 0.1, 0.2, 0.5, 1, 2, 5, 10)
+	c.metrics.NewHistogram("dgraph_mutate_duration", "Response time of Dgraph mutations in milliseconds.",
+		0.05, 0.1, 0.2, 0.5, 1, 2, 5, 10)
+	c.metrics.NewHistogram("dgraph_alter_duration", "Response time of Dgraph alter operations in milliseconds.",
+		0.05, 0.1, 0.2, 0.5, 1, 2, 5, 10)
+
+	dg := dgo.NewDgraphClient(api.NewDgraphClient(conn))
+	c.client = NewDgraphClient(dg)
+
+	// Check connection by performing a basic health check
+	if _, err := c.HealthCheck(context.Background()); err != nil {
+		c.logger.Errorf("dgraph health check failed: %v", err)
+		return
+	}
+}
+
+// UseLogger sets the logger for the Dgraph client which asserts the Logger interface.
+func (c *Client) UseLogger(logger interface{}) {
```

Comment:
```suggestion
func (c *Client) UseLogger(logger any) {
```

Reply:
Replaced `interface{}` with `any` in the entire file